File: arrays-hashing/242_valid_anagram.py
from collections import defaultdict
from typing import Dict

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        count_s: Dict[str, int] = defaultdict(int)
        count_t: Dict[str, int] = defaultdict(int)
        for i in range(len(s)):
            count_s[s[i]] += 1
            count_t[t[i]] += 1
        return count_s == count_t

File: arrays-hashing/242_valid_anagram.cpp
#include <iostream>
#include <array>

class Solution {
public:
    bool isAnagram(std::string s, std::string t) {
        std::array<int, 26> s_counts = {0};
        std::array<int, 26> t_counts = {0};
        for (char c : s) {
            s_counts[c - 'a']++;
        }
        for (char c : t) {
            t_counts[c - 'a']++;
        }
        return s_counts == t_counts;
    }
};
